internal/runtime/client/client.go

```diff
@@ -153,6 +163,107 @@ func (c *client) Unregister(extensionConfig *runtimev1.ExtensionConfig) error {
 	return nil
 }
 
+// CallAllExtensions calls all the ExtensionHandlers registered for the hook.
+// The ExtensionHandler are called sequentially. The function exits immediately after any of the ExtensionHandlers return an error.
+// See CallExtension for more details on when an ExtensionHandler returns an error.
+// The aggregate result of the ExtensionHandlers is updated into the response object passed to the function.
+func (c *client) CallAllExtensions(ctx context.Context, hook catalog.Hook, request runtime.Object, response runtimehooksv1.Response) error {
+	gvh, err := c.catalog.GroupVersionHook(hook)
+	if err != nil {
+		return errors.Wrap(err, "failed to compute GroupVersionHook")
+	}
+	registrations, err := c.registry.List(gvh.GroupHook())
+	if err != nil {
+		return errors.Wrapf(err, "failed to retrieve ExtensionHandlers for %s", gvh.GroupHook())
+	}
+	responses := []runtimehooksv1.Response{}
+	for _, registration := range registrations {
+		responseDescriptor, err := c.catalog.NewResponse(gvh)
+		if err != nil {
+			return errors.Wrap(err, "failed to create response object")
+		}
+
+		tmpResponse := &runtimehooksv1.CommonResponse{
+			TypeMeta: metav1.TypeMeta{Kind: responseDescriptor.GetObjectKind().GroupVersionKind().Kind,
+				APIVersion: responseDescriptor.GetObjectKind().GroupVersionKind().GroupVersion().String()},
+		}
+		err = c.CallExtension(ctx, hook, registration.Name, request, tmpResponse)
+		// If one of the extension handlers fails lets short-circuit here and return early.
+		if err != nil {
+			return errors.Wrapf(err, "ExtensionHandler %s failed", registration.Name)
+		}
+		responses = append(responses, response)
+	}
+	if err := aggregateResponses(responses); err != nil {
+		return errors.Wrap(err, "failed to aggregate responses")
+	}
+	return nil
+}
+
+func aggregateResponses(responses []runtimehooksv1.Response) error {
+	// TODO:(killianmuldoon) implement proper aggregation logic.
```

Comment:
Let's make this a follow-up PR with handling for RetryableResponse, unit tests etc